jax/lax/lax_control_flow.py

```diff
@@ -617,34 +617,81 @@ def _scan_partial_eval(trace, *tracers, **kwargs):
       carry_uk = _map(operator.or_, carry_uk, carry_uk_out)
   else:
     assert False, "Fixpoint not reached"
+  num_res = len(jaxpr_1.out_avals) - len(jaxpr_2.out_avals)
+
+  # The residuals are treated as extensive outputs of jaxpr_1 (and extensive
+  # inputs to jaxpr_2), but residuals that are loop-invariant can be hoisted.
+  # TODO(mattjj): hoist other loop-invariant values here too (instantiate=False)
```

Comment:
I was thinking (vaguely) where you scan a function like `lambda c, _: (5., None)`, i.e. the carry output is constant and can be hoisted out. But that's easy to leave to XLA. Constants in the extensive output seem more interesting because we have more information than XLA (enough to easily produce a broadcast), though those may also not be likely to be created (by humans or transformations).